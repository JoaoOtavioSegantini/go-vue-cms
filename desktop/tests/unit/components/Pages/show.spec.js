import { mount } from "@vue/test-utils";
import ViewPage from "@/components/Pages/View.vue";
import { createStore } from "vuex";

describe("ViewPage.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  const mockRoute = {
    params: {
      id: 1,
    },
  };

  const createVuexStore = () => {
    return createStore({
        state() {
            return {
              Pages: {
                onePage: {
                  ID: "1",
                  title: "Post 1",
                  body: "A simple page text body",
                  slug: "simple-slug-text",
                },
              },
            };
          },
      mutations: {
        updatePage(state, res) {
            state.Pages.onePage = res
        },
      },
      actions: {
        getPage(ctx, res) {
          ctx.commit("updatePage", res);
        },
      },
    });
  };

  function factory() {
    const store = createVuexStore();

    return mount(ViewPage, {
      global: {
        plugins: [store],
        mocks: {
           $route: mockRoute,
        },
      },
    });
  }

  it("renders props.msg when passed", async () => {
    const msg =
      "AÇÕES Post 1 Visualização de página do siteA simple page text bodyEditarRemover";
    const wrapper = factory();

    console.log(wrapper.emitted())

    expect(mockRoute.params.id).toBe(1);
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
