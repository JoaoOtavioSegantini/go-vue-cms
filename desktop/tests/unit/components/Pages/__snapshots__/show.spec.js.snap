// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPage.vue renders props.msg when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/pages\\" class=\\"btn btn-secondary\\" title=\\"Voltar\\"><i class=\\"fa fa-arrow-circle-left\\"></i></a>
    </div>
    <div class=\\"col-11\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Post 1 <small>Visualização de página do site</small></h5>
          <div>A simple page text body</div>
        </div>
        <div class=\\"card-footer\\"><a href=\\"/pages/edit/1\\" class=\\"btn btn-primary\\">Editar</a><a href=\\"/pages/remove/1\\" class=\\"btn btn-danger\\">Remover</a></div>
      </div>
    </div>
  </div>
</div>"
`;
